src/express/server.js

```diff
@@ -224,15 +224,46 @@ class ExpressServer {
         return next(err)
       }
 
+      // do not cache error response
       res.header('Cache-Control', 'no-store')
-      if (_.get(err, 'statusCode') === statusCodeConst.notFound) {
-        res.redirect(`/error/${statusCodeConst.notFound}`)
-      } else {
-        logger.errorReport({
-          report: err,
-          message: 'Error was caught by Express custom error handler.',
-        })
-        res.redirect(`/error/${statusCodeConst.internalServerError}`)
+
+      const errStatusCode = _.get(err, 'statusCode') || _.get(err, 'status')
+
+      switch (errStatusCode) {
+        case statusCodeConst.notFound: {
+          // redirect to 404 error page
+          res.redirect(`/error/${statusCodeConst.notFound}`)
+          return
+        }
+        case statusCodeConst.internalServerError: {
+          // error reporting
+          logger.errorReport({
+            report: err,
+            message: 'Error was caught by Express custom error handler.',
+          })
+          // redirect to 500 error page
+          res.redirect(`/error/${statusCodeConst.internalServerError}`)
+          return
+        }
+        default: {
+          if (errStatusCode < statusCodeConst.internalServerError) {
+            // log client error
+            logger.info(
+              'Client error was caught by Express custom error handler: ',
+              err
+            )
+          } else {
+            // server error report
+            logger.errorReport({
+              report: err,
+              message:
+                'Server error (but not internal server error) was caught by Express custom error handler.',
+            })
+          }
+
+          // TODO: render error page with follow-up guidance to end users
```

Comment:
What would be the follow-up guidance ?

Reply:
Just like 500 or 404 error page, we have instructions to tell end users what's going on now, and we could direct them to other pages or actions.
How do you think?

Reply:
Yeah, that would be great.